src/mesh/index_derivs.cxx

```diff
@@ -1196,7 +1196,7 @@ const Field3D Mesh::indexDDX(const Field3D &f, CELL_LOC outloc, DIFF_METHOD meth
 
       } else if (inloc != CELL_CENTRE) {
         // Interpolate then (centre -> centre) then interpolate
-        return DDX(interp_to(f, CELL_CENTRE), outloc, method);
+        return DDX(interp_to(f, CELL_CENTRE, RGN_NOY), outloc, method);
```

Comment:
This looks like it is changing the standard behaviour, is this the case?

Reply:
I didn't think it changed anything, because I thought the derivative invalidates all the guard cells anyway, so as long as its input has y-guard cells the result should be the same. Is DDX ever expected to give correct results in guard cells?